src/python/WMComponent/JobAccountant/JobAccountantPoller.py

```diff
@@ -70,43 +63,31 @@ def algorithm(self, parameters = None):
         logging.info("Found %d completed jobs" % len(completeJobs))
 
         if len(completeJobs) == 0:
-            # Then we have no work to do.  Bye!
             logging.debug("No work to do; exiting")
             return
 
-        while len(completeJobs) > self.accountantWorkSize:
+        while len(completeJobs) > 0:
             try:
                 jobsSlice = completeJobs[:self.accountantWorkSize]
                 completeJobs = completeJobs[self.accountantWorkSize:]
                 self.accountantWorker(jobsSlice)
+                if len(completeJobs) <= 100:
+                    self.accountantWorker(completeJobs)
+                    break
+                logging.info("Remaining completed jobs to process: %d" % len(completeJobs))
             except WMException:
+                myThread = threading.currentThread()
+                if getattr(myThread, 'transaction', None) != None:
+                    myThread.transaction.rollback()
```

Comment:
Actually it seem it is better to rollback in AccountWorker.py  __call__ method since explicit transaction start and commit happens there